From efbc47b8e43ad173d3319f3f6e87b0a60aad9f37 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 13 Jan 2016 13:43:38 +0100
Subject: [PATCH] network/net-lib.sh: correctly set mac address for
 ip=...:<mtu>:<mac>

if ip=...:<mtu>:<mac> was specified, macadress was not set
---
 modules.d/40network/net-lib.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules.d/40network/net-lib.sh b/modules.d/40network/net-lib.sh
index 256826e..31f1a56 100755
--- a/modules.d/40network/net-lib.sh
+++ b/modules.d/40network/net-lib.sh
@@ -427,6 +427,9 @@ ip_to_var() {
                     ;;
                 [0-9]*)
                     mtu="$8"
+                    if [ -n "${9}" -a -n "${10}" -a -n "${11}" -a -n "${12}" -a -n "${13}" -a -n "${14}" ]; then
+                        macaddr="${9}:${10}:${11}:${12}:${13}:${14}"
+                    fi
                     ;;
                 *)
                     if [ -n "${9}" -a -n "${10}" -a -n "${11}" -a -n "${12}" -a -n "${13}" -a -n "${14}" ]; then
